File: abc157/b/python/b.py
a = [list(map(int, input().split())) for _ in range(3)]
n = int(input())
c = [[0 for _ in range(3)] for _ in range(3)]
flag = 0

for i in range(n):
    b = int(input())
    for j in range(3):
        for k in range(3):
            if a[j][k] == b:
                c[j][k] = 1

for i in range(3):
    if c[i][0] and c[i][1] and c[i][2]:
        flag = 1
    if c[0][i] and c[1][i] and c[2][i]:
        flag = 1

if c[0][0] and c[1][1] and c[2][2]:
    flag = 1
if c[0][2] and c[1][1] and c[2][0]:
    flag = 1

print("Yes") if (flag) else print("No")

File: abc157/b/b.ts
import * as fs from "fs";

function main() {
  const input = fs.readFileSync("/dev/stdin", "utf-8").trim().split(/\s+/).map(Number)
  const a: number[][] = []
  for (let i = 0; i < 3; i++) {
    a.push(input.slice(i * 3, i * 3 + 3))
  }
  const n = input[9]
  const b = input.slice(10)
  let c: number[][] = [
    [0, 0, 0],
    [0, 0, 0],
    [0, 0, 0]
  ]
  let flag = 0

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < 3; j++) {
      for (let k = 0; k < 3; k++) {
        if (a[j][k] == b[i]) {
          c[j][k] = 1;
        }
      }
    }
  }

  for (let i = 0; i < 3; i++) {
    if (c[i][0] && c[i][1] && c[i][2])
      flag = 1;
    if (c[0][i] && c[1][i] && c[2][i])
      flag = 1;
  }
  if (c[0][0] && c[1][1] && c[2][2])
    flag = 1;
  if (c[0][2] && c[1][1] && c[2][0])
    flag = 1;

  if (flag) {
    console.log("Yes")
  }
  else {
    console.log("No")
  }
}

main()
